backend/app/controllers/discussions.js

```diff
@@ -291,6 +291,55 @@ module.exports = {
         });
     },
 
+    /**
+     * Checks if a discussion exists and marks it as resolved
+     * @param {Object} req - Request object
+     * @param {Object} res - Response object
+     * @param {Function} next - Express next middleware function
+     * @return {Object} A json object with either a success or error message
+     */
+    markAsResolved: function (req, res, next) {
+        var thunkQuery = req.thunkQuery;
+        co(function* () {
+
+            // Check if there is a record matching the passed in discussion ID
+            var discussionExist = yield thunkQuery(
+                '( ' +
+                'SELECT count(1) ' +
+                'FROM "Discussions" ' +
+                'WHERE "Discussions"."id" = ' + req.params.id +
```

Comment:
I don't think this will work. This will find a single conversation and mark it as resolved.

When we resolve conversations, we are trying to resolve all the conversations around a given question. So, you need all discussion rows with the same questionId.

Reply:
Oh, my bad. I thought we just wanted to mark one discussion as resolved, but that wouldn't make much sense now that I think of it. I will modify.